### PR TITLE
Fix cross-compilation on osx arm64

### PR DESCRIPTION
**Pull request by traversaro**
See https://github.com/conda-forge/scotch-feedstock/pull/61#issuecomment-1060980663

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -14,46 +18,55 @@ fi
...
+  mv include/metis.h include/scotch/
+  cp -rv include/* $PREFIX/include/
+
+elif [[ "$PKG_NAME" == "ptscotch" ]]; then
+
+  export CCP=mpicc
+  export CCD=${CCP}
+  # This is only used by open-mpi's mpicc
+  # ignored in other cases
+  export OMPI_CC=$CC
```
Comment (isuruf) [inline recipe/build.sh:54]:
```suggestion
  export OMPI_CC=$CC
  export OPAL_PREFIX=$PREFIX
```

Comment (traversaro) [reply recipe/build.sh:54]: Thanks! Just to understand, why is that useful? I tried to check in the `man` manual of `mpicc` (the place where I found about `OMPI_CC`) and there is no reference on `OPAL_PREFIX` .

Comment (isuruf) [reply recipe/build.sh:54]: otherwise ompi will use the internal headers and libraries from `BUILD_PREFIX`

Comment (traversaro) [reply recipe/build.sh:54]: Thanks, applied.

File: recipe/build.sh [hunk 2]
```diff
@@ -14,46 +18,55 @@ fi
...
-# avoid conflicts with the real parmetis.h
-mkdir -p $PREFIX/include/scotch
-cp -v include/parmetis.h  $PREFIX/include/scotch/
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
+  cp $RECIPE_DIR/ptscotch_osx-arm64-$VERSION.h ./src/libscotch/ptscotch.h
+  cp $RECIPE_DIR/ptscotchf_osx-arm64-$VERSION.h ./src/libscotch/ptscotchf.h
+  cp $RECIPE_DIR/scotch_osx-arm64-$VERSION.h ./src/libscotch/scotch.h
+  cp $RECIPE_DIR/scotchf_osx-arm64-$VERSION.h ./src/libscotch/scotchf.h
+  cp $RECIPE_DIR/metis_osx-arm64-$VERSION.h ./src/libscotchmetis/metis.h
+  cp $RECIPE_DIR/parmetis_osx-arm64-$VERSION.h ./src/libscotchmetis/parmetis.h
```
[the remark this answers is not in the document]
Comment (traversaro) [reply recipe/build.sh:27]: See the discussion in https://github.com/conda-forge/scotch-feedstock/pull/61#issuecomment-1046000269, section "dummysize and ptdummysize".

[the remark this answers is not in the document]
Comment (isuruf) [reply recipe/build.sh:27]: Sure, it's not ideal, but it's better than hardcoding stuff like this. In my experience the two platforms are almost identical in sizes (except for double) that running a dummysize for intel will mostly work for ptdummysize. (We do similar for many feedstocks)
Can you check if the two headers match for arm64 vs intel?

My concern here is the maintainability of this. This hardcoding is going to introduce a lot of pain when updating the recipe.

[the remark this answers is not in the document]
Comment (traversaro) [reply recipe/build.sh:27]: Let's move the discussion back in the original PR: https://github.com/conda-forge/scotch-feedstock/pull/61#issuecomment-1061082304 .

**Pull request closed without merging**